pkg/util/httpclient/httpclient.go

```diff
@@ -0,0 +1,75 @@
+package httpclient
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"flag"
+	"io/ioutil"
+	"net/url"
+	"time"
+
+	"github.com/prometheus/common/log"
+
+	"github.com/cortexproject/cortex/pkg/util/flagext"
+)
+
+// Config is the config for the HTTP client
+type Config struct {
+	Endpoint           flagext.URLValue `yaml:"endpoint"`
+	ClientTimeout      time.Duration    `yaml:"client_timeout"`
+	BackendReadTimeout time.Duration    `yaml:"backend_read_timeout"`
```

Comment:
This is used only by the query-tee. If there's a config option in a generic `Config` then it must be supported by any client using this config. In this specific case, I would suggest to rollback any change in the `query-tee` (not supporting TLS there is fine, the query-tee is just used for testing right now and we could add TLS support anytime in the future) and I would remove this config option at all.